src/modules/service-catalog/components/useItemFormFields.spec.ts

```diff
@@ -0,0 +1,394 @@
+
+describe("useItemFormFields", () => {
+  const mockedItem: ServiceCatalogItem = {
+    id: 1,
+    name: "Test Item",
+    description: "Test Description",
+    form_id: 1,
+  };
+
+  const textField = {
+    id: 1,
+    type: "text",
+    description: "Test Description",
+    title_in_portal: "Test Title",
+    editable_in_portal: true,
+    required_in_portal: true,
+  };
+
+  const lookupField = {
+    id: 2,
+    type: "lookup",
+    description: "Service",
+    title_in_portal: "Service",
+    editable_in_portal: true,
+    relationship_target_type: "standard::service_catalog_item",
+    required_in_portal: true,
+  };
+
+  const expectedTextField = {
+    id: 1,
+    type: "text",
+    name: "custom_fields_1",
+    description: "Test Description",
+    label: "Test Title",
+    options: undefined,
+    required: true,
+    relationship_target_type: undefined,
+    error: null,
+    value: null,
+  };
+
+  const expectedLookupField = {
+    description: "Service",
+    editable_in_portal: true,
+    id: 2,
+    relationship_target_type: "standard::service_catalog_item",
+    required_in_portal: true,
+    title_in_portal: "Service",
+    type: "lookup",
+  };
+
+  const additionalTextField = {
+    id: 3,
+    type: "text",
+    name: "custom_fields_3",
+    description: "Test Description 3",
+    label: "Test Title 3",
+    options: [],
+    required: true,
+    relationship_target_type: undefined,
+    error: null,
+    value: "Test Value 3",
+  };
+
+  const mockedBaseLocale = "en-us";
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should initialize with default values", async () => {
+    const mockedFormResponse = {
+      ticket_form: {
+        id: 1,
+        ticket_field_ids: [1, 2],
+        active: true,
+      },
+    };
+
+    const mockedTicketFieldResponse = {
+      ticket_fields: [textField, lookupField],
+    };
+
+    global.fetch = jest.fn((url) => {
+      if (url.includes("/ticket_forms")) {
+        return Promise.resolve({
+          json: () => Promise.resolve(mockedFormResponse),
+          status: 200,
+          ok: true,
+        });
+      } else {
+        return Promise.resolve({
+          json: () => Promise.resolve(mockedTicketFieldResponse),
+          status: 200,
+          ok: true,
+        });
+      }
+    }) as jest.Mock;
+
+    const { result, waitForNextUpdate } = renderHook(() =>
+      useItemFormFields(mockedItem, mockedBaseLocale)
+    );
+
+    expect(result.current.requestFields).toEqual([]);
+    expect(result.current.associatedLookupField).toBe(undefined);
+    expect(result.current.error).toBe(null);
+
+    await waitForNextUpdate();
+
+    expect(result.current.requestFields).toEqual([expectedTextField]);
+
+    expect(result.current.associatedLookupField).toEqual(expectedLookupField);
+    expect(result.current.error).toBe(null);
+  });
+
+  it("should return an error if form data fetch fails", async () => {
+    global.fetch = jest.fn(() =>
```

Comment:
Nit: We could also "make it" a `jest.Mock` before the tests and then just mock the resolved value when needed:

``` ts
const globalFetch = ((globalThis.fetch as jest.Mock) = jest.fn());

//...

globalFetch.mockResolvedValue({ status: 500, ok: false })

```